packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -395,7 +395,25 @@ export class BaseSnapExecutor {
       }
     };
 
-    return { request: request as RequestFunction };
```

Comment:
if we don't have the need to wrap the stream provider, hardening this plain object would have been enough. Is there a plan to add more complex functionality to the proxy here?